### PR TITLE
Feature/restore password

### PR DESCRIPTION
**Pull request by AKutusenko**
## Describe your changes

- I have added restore password page and verify code page. Also i have added api to send verification code and check is the code valid.

image

image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-45)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [ ] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/sections/restore-password-view/form.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import Stack from '@mui/system/Stack';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from 'store';
+import { patterns } from 'constants/patterns';
+import FormProvider from 'components/hook-form';
+import { useSendCodeMutation } from 'store/api/restorePasswordApi';
+import { setEmail } from 'store/reducers/restorePasswordReducer';
+import { links } from 'constants/links';
+
+export default function RestorePasswordForm() {
```
[the remark this answers is not in the document]
Comment (AKutusenko) [reply src/sections/restore-password-view/form.tsx]: i have fixed it

File: src/sections/restore-password-view/form.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  const t = useTranslations('RestorePasswordPage');
+  const [sendCode, { isLoading }] = useSendCodeMutation();
+  const router = useRouter();
+  const dispatch = useDispatch<AppDispatch>();
+  const { enqueueSnackbar } = useSnackbar();
+  const defaultValues = { email: '' };
```
[the remark this answers is not in the document]
Comment (AKutusenko) [reply src/sections/restore-password-view/form.tsx]: i have fixed it

File: src/sections/verify-code-view/form.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import { links } from 'constants/links';
+
+export default function RestorePasswordForm() {
+  const t = useTranslations('VerifyCodePage');
+  const [verifyCode, { isLoading }] = useVerifyCodeMutation();
+  const [sendCode] = useSendCodeMutation();
+  const router = useRouter();
+  const dispatch = useDispatch<AppDispatch>();
+  const { enqueueSnackbar } = useSnackbar();
+  const defaultValues = { code: '' };
```
[the remark this answers is not in the document]
Comment (AKutusenko) [reply src/sections/verify-code-view/form.tsx]: i have fixed it

File: src/sections/verify-code-view/form.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  const {
+    reset,
+    formState,
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+  const { isDirty, isValid } = formState;
+
+  const onSubmit = handleSubmit(async (data) => {
+    const EMAIL_ACTIVATED = 'Email already activated';
```
[the remark this answers is not in the document]
Comment (AKutusenko) [reply src/sections/verify-code-view/form.tsx]: i have fixed it

File: src/sections/verify-code-view/form.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    try {
+      await new Promise((resolve) => setTimeout(resolve, 3000));
+      reset();
+      const payload = { email, code: data.code };
+      const response = await verifyCode(payload);
+      if (response.error.data.message === EMAIL_ACTIVATED) {
```
[the remark this answers is not in the document]
Comment (AKutusenko) [reply src/sections/verify-code-view/form.tsx]: i have fixed it

**Review by lyudad — APPROVED** [review]

**Pull request merged**